mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,589 @@
+# -*- coding: utf-8 -*-
+# This software is open source software available under the BSD-3 license.
+#
+# Copyright (c) 2022 Triad National Security, LLC. All rights reserved.
+# Copyright (c) 2022 Lawrence Livermore National Security, LLC. All rights
+# reserved.
+# Copyright (c) 2022 UT-Battelle, LLC. All rights reserved.
+#
+# Additional copyright and license information can be found in the LICENSE file
+# distributed with this code, or at
+# https://raw.githubusercontent.com/MPAS-Dev/MPAS-Analysis/master/LICENSE
+#
+import os
+import xarray
+import numpy
+import matplotlib.pyplot as plt
+
+from mpas_analysis.shared import AnalysisTask
+
+from mpas_analysis.shared.io import open_mpas_dataset, write_netcdf
+from mpas_analysis.shared.io.utility import build_config_full_path, \
+    build_obs_path, make_directories, decode_strings
+from mpas_analysis.shared.climatology import compute_climatology, \
+    get_unmasked_mpas_climatology_file_name
+
+from mpas_analysis.shared.constants import constants
+from mpas_analysis.shared.plot import histogram_analysis_plot, savefig
+from mpas_analysis.shared.html import write_image_xml
+
+
+class OceanHistogram(AnalysisTask):
+    """
+    Plots a histogram of a 2-d ocean variable.
+
+    """
+
+    def __init__(self, config, mpasClimatologyTask, regionMasksTask,
+                 controlConfig=None):
+
+        """
+        Construct the analysis task.
+
+        Parameters
+        ----------
+        config : mpas_tools.config.MpasConfigParser
+            Configuration options
+
+        mpasClimatologyTask : ``MpasClimatologyTask``
+            The task that produced the climatology to be remapped and plotted
+
+        regionMasksTask : ``ComputeRegionMasks``
+            A task for computing region masks
+
+        controlConfig : mpas_tools.config.MpasConfigParser
+            Configuration options for a control run (if any)
+        """
+
+        # first, call the constructor from the base class (AnalysisTask)
+        super().__init__(
+            config=config,
+            taskName='oceanHistogram',
+            componentName='ocean',
+            tags=['climatology', 'regions', 'histogram', 'publicObs'])
+
+        self.run_after(mpasClimatologyTask)
+        self.mpasClimatologyTask = mpasClimatologyTask
+
+        self.controlConfig = controlConfig
+        mainRunName = config.get('runs', 'mainRunName')
+
+        self.regionGroups = config.getexpression(self.taskName, 'regionGroups')
+        self.regionNames = config.getexpression(self.taskName, 'regionNames')
+        self.seasons = config.getexpression(self.taskName, 'seasons')
+        self.variableList = config.getexpression(self.taskName, 'variableList')
+        if config.has_option(self.taskName, 'weightList'):
+            self.weightList = config.getexpression(self.taskName, 'weightList')
+        else:
+            self.weightList = None
+
+        self.filePrefix = f'histogram_{mainRunName}'
```

Comment:
I imagine there is some confusion here about main vs. control type runs but I don't think there's a need to have the run name in the file name.  Each analysis run goes into its own directory so the files don't need to have that info in them.

Instead, I do think it is necessary for the file name to have the `regionGroup` but with spaces replaced by underscores.  I don't think you need `self.filePrefix`, since this attribute won't get used anywhere outside this function (it's not used in `setup_and_check()`).  Instead, you can move this into the `regionGroups` loop and just have a local variable:
```suggestion
        regionGroupSuffix = regionGroup.replace(' ', '_')
        filePrefix = f'histogram_{regionGroupSuffix}'
```